### PR TITLE
Throw InvalidParam for unexpected input of Params::decode()

### PR DESCRIPTION
**Pull request by njam**
@dlondero @fvovan please review

### CODE REVIEW

File: library/CM/Params.php [hunk 1]
```diff
@@ -582,16 +582,21 @@ public static function encodeObjectId(CM_ArrayConvertible $object) {
...
-            // CM_ArrayConvertible
             $className = (string) $value['_class'];
+            if (!class_exists($className)) {
+                throw new CM_Exception_InvalidParam('Class for decoding does not exist', null, ['class' => $className]);
+            }
+            if (!is_subclass_of($className, 'CM_ArrayConvertible')) {
+                throw new CM_Exception_InvalidParam('Class for decoding is not CM_ArrayConvertible', null, ['class' => $className]);
+            }
             unset($value['_class']);
             $value = call_user_func(array($className, 'fromArray'), $value);
```
Comment (fvovan) [inline library/CM/Params.php]: since we did all checking is it possible to use `$className::fromArray($value)` here  ?

**Comment by njam** [comment]
yes, please review

**Comment by njam** [comment]
@dlondero @fvovan please review

**Comment by fvovan** [comment]
lgtm

**Pull request merged**